integration/experiment/power_sweep/gen_policy_recommendation.py

```diff
@@ -0,0 +1,287 @@
+from sklearn.linear_model import LinearRegression
+
+from geopmpy import io as geoio
+
+# abstract model class
+# TODO add types and help text! modern python!
+class PowerLimitModel:
+    def __init__(self):
+        # TODO raise not implemented
+        pass
+
+    def train(self, PL_list, value_list):
+        # TODO raise not implemented
+        pass
+
+    def test(self, PL):
+        # TODO raise not implemented
+        pass
+
+    def batch_test(self, PL_list):
+        return [self.test(PL) for PL in PL_list]
+
+    def serialize(self):
+        # TODO raise not implemented
+        # actually make this convert to/from json nicely
+        pass
+
+    def __str__(self):
+        # TODO classname?
+        return "<%s>" % "PowerLimitModel"
+
+    def repr(self):
+        return self.__str__()
+
+
+class PolynomialFitPowerModel(PowerLimitModel):
+    def __init__(self, degree):
+        self._degree = degree
+        self._transform = PolynomialFeatures(
+                degree=self._degree,
+                include_bias=False)
+        self._model = None
+
+    def train(self, PL_list, value_list):
+        # TODO check if PL_list and value_list are the same length
+        x1 = np.array(PL_list).reshape((-1, 1))
+        yy = np.array(value_list)
+        xn = self._transform.fit_transform(x1)
+        self._model = LinearRegression().fit(xn, yy)
+
+    def test(self, PL):
+        # TODO shouldn't have to look inside the coeffs and 
+        # intercept to do this
+        return self._model.intercept_ + \
+               sum([
+                   c * PL**(i+1) for i, c in enumerate(self._model.coef_)
+                   ])
+
+    def __str__(self):
+        if self._model:
+            terms = ["", " PL"] + \
+                    [" PL ** %d" % (i+1)
+                            for i in range(1, len(self._model.coef_))]
+            coeffs = [self._model.intercept] + self._model.coef_
+            expression = ""
+            terms.reverse() ; coeffs.reverse() ; firstTerm = True
+            for tt, cc in zip(terms, coeffs):
+                if not firstTerm:
+                    if cc < 0:
+                        expression += " - "
+                        cc *= -1
+                    else:
+                        expression += " + "
+                expression += str(cc) + tt
+            # TODO classname
+            return "<%s %s>" % ("Polyfitmodel", expression)
+        else:
+            return "<%s (untrained)>" % "Polyfitmodel"  #TODO classname
+
+
+class CubicPowerModel(PolynomialFitPowerModel):
+    def __init__(self):
+        super().__init__(degree=3)
+
+
+def extract_from_reports(path, region_filter):
+    simplified_data = []
+    
+    try:
+        df = geoio.RawReportCollection("*report", dir_name=path).get_df()
+    except RuntimeError:
+        sys.stderr.write("<geopm> Error: No report data found in " + path + \
+                "; run a power sweep before using this analysis.\n")
+        sys.exit(1)
```

Comment:
Please refactor this so the initial report parsing happens in the ```if __name__``` block at the bottom.  Use  https://github.com/geopm/geopm/blob/dev/integration/experiment/power_sweep/gen_plot_balancer_comparison.py as a reference.

Reply:
Done.